ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -101,22 +101,13 @@ implAddTxs mpEnv@MempoolEnv{mpEnvStateVar, mpEnvLedgerCfg} txs =
 implGetTxs :: (MonadSTM m, ApplyTx blk)
            => MempoolEnv m blk hdr
            -> STM m (Seq (GenTx blk, TicketNo))
-implGetTxs mpEnv@MempoolEnv{mpEnvStateVar} = do
-    ValidationResult {
-      vrBefore,
-      vrValid
-    } <- validateIS mpEnv
-    -- TODO (maybe): log the transactions that we silently discard
-    writeTVar mpEnvStateVar IS { isTxs = vrValid
-                               , isTip = vrBefore
-                               }
-    pure $ fromTxSeq vrValid
+implGetTxs = (flip implGetTxsAfter) zeroTicketNo
```

Comment:
Then we need a QC property that asserts that adding a list of txs to a mempool and then getting them retrieves the same list. Again, this ought to catch a 0 vs 1 index bug.

That said, perhaps it's better to keep the original, since that doesn't incur the O(log n) search cost, and this operation does happen in the hot path of block creation.

Something we could consider for the future here: we could also augment the `TxSeq` finger tree with the size in bytes of the txs, and that would allow the block creation code to select the first N transactions that are less than a given byte limit (ie block size) and do so in log time rather than linear time. This means shifting work: doing less when creating blocks, and a little more elsewhere to maintain the extra finger tree  meaure monoid. File a ticket so we remember to consider it later.